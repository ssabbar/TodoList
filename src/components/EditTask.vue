<template>
  <div>
    <div>
      <p class="mb-2">Edit name for {{task.name}}</p>
      <input class="mb-3" v-model="taskNewName"/>
    </div>
    <div class="row">
      <button @click="removeEditMode()" id="cancelEditButton" class="btn btn-outline-secondary">Cancel</button>
      <button @click="saveEditTask(taskNewName)" id="saveEditButton" class="btn btn-dark">Save</button>
    </div>
  </div>
</template>

<script>
export default {
  name: 'EditTask',
  data () {
    return {
      taskNewName: this.task.name
    }
  },
  props: {
    task: Object
  },
  methods: {
    removeEditMode () {
      this.$emit('updateEditedValue', false)
    },
    saveEditTask (taskNewName) {
      this.$store.dispatch({
        type: 'saveEditTask',
        taskNewName: taskNewName,
        taskId: this.task.id
      })
      this.removeEditMode()
    }
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
button{
  display: inline-block;
  width: 45%;
  margin-left: 1rem;
}

input {
  width: 100%;
}

p {
  font-size: large;
}
</style>
