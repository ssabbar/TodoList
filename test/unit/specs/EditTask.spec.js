import Vue from 'vue'
import Vuex from 'vuex'
import { mount, createLocalVue } from '@vue/test-utils'
import EditTask from '@/components/EditTask'

const localVue = createLocalVue()

localVue.use(Vuex)

describe('EditTask component', () => {
  let actions
  let store

  const mockedPropData = {
    task: {
      id: 1,
      name: 'task1',
      completed: false
    }
  }

  beforeEach(() => {
    actions = {
      saveEditTask: jest.fn()
    }
    store = new Vuex.Store({
      actions
    })
  })

  it('renders correctly', () => {
    const wrapper = mount(EditTask, { propsData: { ...mockedPropData } })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('call store action "saveEditTask" when click on save button', async () => {
    const wrapper = mount(EditTask, {
      store,
      localVue,
      propsData: { ...mockedPropData }
    })

    wrapper.find('input').setValue('newTask')
    await wrapper.find('#saveEditButton').trigger('click')
    expect(actions.saveEditTask)
      .toHaveBeenCalledWith(
        expect.anything(),
        {
          type: 'saveEditTask',
          taskNewName: 'newTask',
          taskId: 1
        })

    expect(wrapper.emitted().updateEditedValue[0][0]).toBe(false)
  })
})
