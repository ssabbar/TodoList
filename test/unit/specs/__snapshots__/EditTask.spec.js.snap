// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditTask component renders correctly 1`] = `
<div>
    <div>
        <p class="mb-2">Edit name for task1</p> <input class="mb-3">
    </div>
    <div class="row"><button id="cancelEditButton" class="btn btn-outline-secondary">Cancel</button> <button id="saveEditButton" class="btn btn-dark">Save</button></div>
</div>
`;
